shopfloor_reception/services/reception.py

```diff
@@ -0,0 +1,1079 @@
+        search = self._actions_for("search")
+        packaging = search.packaging_from_scan(barcode)
+        if packaging:
+            selected_line.qty_done += packaging.qty
+            return self._response_for_set_quantity(picking, selected_line)
+
+    def _set_quantity__by_package(self, picking, selected_line, barcode):
+        search = self._actions_for("search")
+        package = search.package_from_scan(barcode)
+        if package:
+            dest_location = selected_line.location_dest_id
+            child_locations = self.env["stock.location"].search(
+                [("id", "child_of", dest_location.id)]
+            )
+            pack_location = package.location_id
+            if pack_location:
+                if pack_location not in child_locations:
+                    # If the scanned package has a location that isn't a child
+                    # of the move dest, return an error
+                    message = self.msg_store.dest_location_not_allowed()
+                    return self._response_for_set_quantity(
+                        picking, selected_line, message=message
+                    )
+                else:
+                    # If the scanned package has a valid destination,
+                    # set both package and destination on the package,
+                    # and go back to the selection line screen
+                    selected_line.result_package_id = package
+                    selected_line.location_dest_id = pack_location
+                    return self._response_for_select_line(picking)
+            # Scanned package has no location, move to the location selection
+            # screen
+            selected_line.result_package_id = package
+            return self._response_for_set_destination(picking, selected_line)
+
+    def _set_quantity__by_location(self, picking, selected_line, barcode):
+        search = self._actions_for("search")
+        location = search.location_from_scan(barcode)
+        if location:
+            dest_location = selected_line.location_dest_id
+            child_locations = self.env["stock.location"].search(
+                [("id", "child_of", dest_location.id)]
+            )
+            if location not in child_locations:
+                # Scanned location isn't a child of the move's dest location
+                message = self.msg_store.dest_location_not_allowed()
+                return self._response_for_set_quantity(
+                    picking, selected_line, message=message
+                )
+            # process without pack, set destination location, and go back to
+            # `select_line`
+            selected_line.location_dest_id = location
+            return self._response_for_select_line(picking)
+
+    def _use_handlers(self, handlers, *args, **kwargs):
+        for handler in handlers:
+            response = handler(*args, **kwargs)
+            if response:
+                return response
```

Comment:
```suggestion
            return handler(*args, **kwargs)
```

Reply:
Hmmm wouldn't this always return whatever the first handler returns, and stop the loop there?